crucible-llvm/src/Lang/Crucible/LLVM/Translation/Instruction.hs

```diff
@@ -1879,45 +1883,53 @@ callFunction instr _tailCall fnTy _fn _args _assign_f =
 -- instructions.
 callFunctionWithCont :: forall s arch ret a.
    (?transOpts :: TranslationOptions) =>
-   L.Instr {- ^ Source instruction o the call -} ->
+   Set L.Ident {- ^ Set of usable identifiers -} ->
+   L.Instr {- ^ Source instruction of the call -} ->
    Bool    {- ^ Is the function a tail call? -} ->
    L.Type  {- ^ type of the function to call -} ->
    L.Value {- ^ function value to call -} ->
    [L.Typed L.Value] {- ^ argument list -} ->
    (LLVMExpr s arch -> LLVMGenerator s arch ret ()) {- ^ assignment continuation for return value -} ->
    LLVMGenerator s arch ret a {- ^ continuation for next instructions -} ->
    LLVMGenerator s arch ret a
-callFunctionWithCont instr tailCall_ fnTy fn args assign_f k
+callFunctionWithCont defSet instr tailCall_ fnTy fn args assign_f k
 
      -- Supports LLVM 4-12
-     | L.ValSymbol "llvm.dbg.declare" <- fn =
-       do mbArgs <- dbgArgs args
+     | L.ValSymbol "llvm.dbg.declare" <- fn
+     , debugIntrinsics ?transOpts =
+       do mbArgs <- dbgArgs defSet args
           case mbArgs of
             Right (asScalar -> Scalar _ PtrRepr ptr, lv, di) ->
               extensionStmt (LLVM_Debug (LLVM_Dbg_Declare ptr lv di)) >> k
+            Left msg -> addWarning (Text.pack msg) >> k
             _ -> k
```

Comment:
I know this technically isn't your fault, but having these catch-call `_ -> k` cases is a bit worrisome, as it's possible that we could fall through silently if given unusual debug information (which, as we've seen, is totally within the realm of possibility). Perhaps we should give a warning here at least?